### DISCUSSION TITLE
v3 needs wired model to be exact boolean in input checkbox

### DISCUSSION THREAD

**Discussion by almazdanilov** [livewire/livewire#6905] [7. Bugs (archived)]
### Livewire version

v3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

Laravel reads and writes boolean values as tinyint(1) into/from MySql database. Livewire v2 understood that '1' is 'true'. But v3 now needs boolean to be real 'true' value to make checkbox checked. There is a solution to write $casts to bool in model class or use 'legacy_model_binding' => true. I believe it is bug and '1' will be used by Livewire v3 as 'true' in future.

### Please confirm (incomplete submissions will not be addressed)

- [x] I have provided easy and step-by-step instructions to reproduce the bug.
- [x] I have provided code samples as text and NOT images.
- [x] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ITSYV** [comment]
Another solution seems to set the checkbox as 'checked="true"' by default. Livewire unchecks it when it needs to or just keeps it checked if the value is set.
This works when using other values than just booleans (json storing for example)

**Comment by CyberPunkCodes** [comment]
I'm surprised this hasn't been solved yet. Just stumbled across it.

FWIW: I was not able to get casting in my model to work.

I have to explicitly set it to true/false before it hits the livewire form:

```php
$this->is_valid_email = ($wordpressUser->is_valid_email == 1) ? true : false;
```

**Comment by almazdanilov** [comment]
This error has not yet been fixed in v3.4.4.
@calebporzio, please clarify, will this bug be fixed or do you think it is not the bug?
This new strict boolean behavior is also conflicting with the official v3 docs at
https://livewire.laravel.com/docs/wire-model#checkboxes 
which says that we can use different values in checkboxes:
```
<input type="checkbox" value="email" wire:model="updateTypes">
<input type="checkbox" value="sms" wire:model="updateTypes">
<input type="checkbox" value="notification" wire:model="updateTypes">
```
But this example is not working because value of checkbox in v3 can only be true or false.
Can you please correct the documentation if not planning to correct this strange behavior of livewire?

**Reply by gingebaker** [reply]
I just checked the Livewire JS wich I think is responsible for setting the checked status initially:
```js
//...
else if (el.type === "checkbox") {
	if (Number.isInteger(value)) {
		el.value = value;
	} else if (!Array.isArray(value) && typeof value !== "boolean" && ![null, void 0].includes(value)) {
		el.value = String(value);
	} else {
		if (Array.isArray(value)) {
			el.checked = value.some((val) => checkedAttrLooseCompare(val, el.value));
		} else {
			el.checked = !!value;
		}
	}
}
```

So I guess there is no possibility for now to get the desired behaviour vor model on checkbox to get the checked attribute enabled when the value is "1" or "0" as number or string? 
Would be interesting why this was changed in Livewire3? I can´t see the benefit for now?

**Comment by lsmith77** [comment]
I would also appreciate if this would be handled just like in Livewire 2. Don't really see the benefit of checking for boolean for scripting languages.

I tried the `checked="true"` hack but it doesn't work for me.

Also tried `.boolean` which also didn't do the trick :-/

**Comment by ITSYV** [comment]
I just submitted a failing test PR for this problem. (https://github.com/livewire/livewire/pull/8291) Hope to get some clarification soon about this

**Comment by manhtruongwang** [comment]
``` php
else if (el.type === "checkbox") {
        if(Array.isArray(value)){
          var valueFound=!1;
          value.forEach((function(val){
            val==el.value&&(valueFound=!0)
          })),
          el.checked=valueFound
        } else {
          el.checked=!!value; 
        }
      }
```

I have to replace livewire.esm.js with the livewire 2 and it now working as expected

**Comment by Alletkla** [comment]
This is very annoying since on my side wire.model.live in combination with #[URL] usage the checkbox values are getting strings. 
And therefore when read back in the checkbox isn't checked. Very annoying and needs to be fixed!
